src/main/drivers/rx/rx_sx1280.c

```diff
@@ -338,58 +342,136 @@ uint8_t sx1280GetStatus(void)
 {
-    sx1280SetMode(SX1280_MODE_SLEEP);
-    sx1280PollBusy();
+    uint8_t rfparams[3];
+    rfparams[0] = (uint8_t)sf;
+    rfparams[1] = (uint8_t)bw;
+    rfparams[2] = (uint8_t)cr;
 
-    sx1280ConfigLoraDefaults();
-    sx1280SetOutputPower(13); //default is max power (12.5dBm for SX1280 RX)
-    sx1280SetMode(SX1280_MODE_STDBY_RC); 
-    sx1280ClearIrqStatus(SX1280_IRQ_RADIO_ALL);
-    sx1280ConfigLoraModParams(bw, sf, cr);
-    sx1280SetPacketParams(preambleLength, SX1280_LORA_PACKET_IMPLICIT, 8, SX1280_LORA_CRC_OFF, (sx1280LoraIqModes_e)((uint8_t)!iqInverted << 6)); // TODO don't make static etc.
-    sx1280SetFrequencyReg(freq);
-}
+    sx1280WriteCommandBurst(SX1280_RADIO_SET_MODULATIONPARAMS, rfparams, 3);
 
-void sx1280SetOutputPower(const int8_t power)
-{
-    uint8_t buf[2];
-    buf[0] = power + 18;
-    buf[1] = (uint8_t) SX1280_RADIO_RAMP_04_US;
-    sx1280WriteCommandBurst(SX1280_RADIO_SET_TXPARAMS, buf, 2);
+    switch (sf) {
+    case SX1280_LORA_SF5:
+    case SX1280_LORA_SF6:
+        sx1280WriteRegister(SX1280_REG_SF_ADDITIONAL_CONFIG, 0x1E); // SF5 or SF6
+        break;
+    case SX1280_LORA_SF7:
+    case SX1280_LORA_SF8:
+        sx1280WriteRegister(SX1280_REG_SF_ADDITIONAL_CONFIG, 0x37); // SF7 or SF8
+        break;
+    default:
+        sx1280WriteRegister(SX1280_REG_SF_ADDITIONAL_CONFIG, 0x32); // SF9, SF10, SF11, SF12
+    }
 }
 
-void sx1280SetPacketParams(const uint8_t preambleLength, const sx1280LoraPacketLengthsModes_e headerType, const uint8_t payloadLength, 
-                           const sx1280LoraCrcModes_e crc, const sx1280LoraIqModes_e invertIQ)
+static void sx1280SetPacketParamsLora(const uint8_t preambleLength, const sx1280LoraPacketLengthsModes_e headerType, const uint8_t payloadLength,
+    const sx1280LoraCrcModes_e crc, const bool invertIQ)
 {
     uint8_t buf[7];
-
     buf[0] = preambleLength;
     buf[1] = headerType;
     buf[2] = payloadLength;
     buf[3] = crc;
-    buf[4] = invertIQ;
+    buf[4] = invertIQ ? SX1280_LORA_IQ_INVERTED : SX1280_LORA_IQ_NORMAL;
     buf[5] = 0x00;
     buf[6] = 0x00;
 
     sx1280WriteCommandBurst(SX1280_RADIO_SET_PACKETPARAMS, buf, 7);
 }
 
+static void sx1280ConfigModParamsFlrc(const SX1280_RadioFlrcBandwidths_t bw, const SX1280_RadioFlrcCodingRates_t cr, const SX1280_RadioFlrcGaussianFilter_t bt)
+{
+    uint8_t rfparams[3];
+    rfparams[0] = (uint8_t)bw;
+    rfparams[1] = (uint8_t)cr;
+    rfparams[2] = (uint8_t)bt;
+
+    sx1280WriteCommandBurst(SX1280_RADIO_SET_MODULATIONPARAMS, rfparams, 3);
+}
+
+static void sx1280SetPacketParamsFlrc(uint8_t PreambleLength, uint8_t HeaderType,
+    uint8_t PayloadLength, uint32_t syncWord, uint16_t crcSeed, uint8_t cr)
+{
+    if (PreambleLength < 8)
+        PreambleLength = 8;
+    PreambleLength = ((PreambleLength / 4) - 1) << 4;
+
+    uint8_t buf[7];
+    buf[0] = PreambleLength;                    // AGCPreambleLength
```

Comment:
I did not change this to use the register value since that would require code to convert from any given length to the proper register and, again, let's keep the code the same as ExpressLRS without trying to improve on it independently here.